### PR TITLE
Document routing constraints

### PR DESCRIPTION
**Pull request by derekprior**

### CODE REVIEW

File: lib/clearance/constraints/signed_in.rb [hunk 1]
```diff
@@ -1,5 +1,22 @@
...
+    #       resources :posts
+    #     end
+    #
+    # In the example above, requests to `/posts` from users that are not signed
+    # in, will result in a 404. You can make additional assertions about the
+    # user by passing a block. For instance, if you want to require that the
+    # signed-in user be an admin:
+    #
+    #     # config/routes.rb
+    #     constraints Clearance::Constraints::SignedIn.new { |user| user.admin? } do
```
Comment (derekprior) [inline lib/clearance/constraints/signed_in.rb]: Heel!

Comment (sgrif) [inline lib/clearance/constraints/signed_in.rb]: Is it worth suggesting that users extract that to a method or local variable when they're adding additional stuff?

File: lib/clearance/constraints/signed_in.rb [hunk 2]
```diff
@@ -1,5 +1,22 @@
...
+    # Can be applied to make a set of routes visible only to users that are
+    # signed in.
+    #
+    #     # config/routes.rb
+    #     constraints Clearance::Constraints::SignedIn.new do
+    #       resources :posts
+    #     end
+    #
+    # In the example above, requests to `/posts` from users that are not signed
+    # in, will result in a 404. You can make additional assertions about the
```
Comment (gabebw) [inline lib/clearance/constraints/signed_in.rb]: I'd remove this comma.

File: lib/clearance/constraints/signed_out.rb [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
+    # Can be applied to make a set of routes visible only to users that are
+    # signed out.
+    #
+    #     # config/routes.rb
+    #     constraints Clearance::Constraints::SignedOut.new do
+    #       resources :registrations, only: [:new, :create]
+    #     end
+    #
+    # In the example above, requests to `/registrations/new` from users that are
+    # already signed in will result in a 404.
```
Comment (gabebw) [inline lib/clearance/constraints/signed_out.rb]: I'd remove "already" here.

File: lib/clearance/constraints/signed_in.rb (hunk 1 above)
Comment (derekprior) [inline lib/clearance/constraints/signed_in.rb]: every local variable extraction I tried that made sense to me was longer than the current line. So in the interest of keeping the example simple I'm just going with whats here.

**Comment by gabebw** [comment]
:+1:

**Pull request merged**